Search page and functionality

MarkGuo2000 opened the pull request:
Adding Search page where the using type the name of the author or post content. Filtering done though matching content and search terms.

# Description:
Please provide a summary of changes made and the relevant issue that has been worked on. Also list any major changes made to project structure/dependencies if applicable

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](../CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project
- [ ] My code has been commented
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

MarkGuo2000 committed:
Search page and functionality

Adding Search page where the using type the name of the author or post content. Filtering done though matching content and search terms.

hmit106 committed:
Update Jest to 27.5.1 (#132)

hmit106 commented on frontend/src/Router.jsx:
```diff
@@ -30,6 +32,7 @@ const Router = () => {
         <Route path="/notifications" element={<Notifications />} />
         <Route path="/settings" element={<ProfileSettings />} />
         <Route path="/new-post" element={<PostComposer />} />
+        <Route path="/SeachPage" element={<SearchPage/>} />
```
Watch out for the typo here - says "SeachPage"

Michaelrecycle replied:
path names are kebab case for this one

abis555 committed:
[#82] Added UPDATE and DELETE function for user endpoint (#115)

* Added UPDATE and DELETE function for user endpoint

* Fixed issue#82

* [SE701Team2#82] - Added UPDATE and DELETE endpoints for user

* [SE701Team2#82] - Added UPDATE and DELETE endpoints for user

* [SE701Team2#82] - Added UPDATE and DELETE endpoints for user

* updated

* updated

* updated

* updated

* updated

* updated

* updated

* [SE701Team2#82] - Added UPDATE and DELETE endpoints for user

* updated

* updated

* [#82] - Added UPDATE and DELETE functions for user endpoint

Michaelrecycle commented on frontend/src/pages/post/SearchPage.jsx:
```diff
@@ -0,0 +1,25 @@
+
+    return (
+        <div className="Search">
+            <input type="text" placeholder="Seach..." onChange={event => {setSeachTerm(event.target.value);}} />
```
use the MUI text input component, they probably have search field templates

Mingyang-Li replied:
They do

nzbasic commented on frontend/src/pages/post/SearchPage.jsx:
```diff
@@ -0,0 +1,25 @@
+import { useParams } from "react-router-dom"
+import PostPageView from "./PostPageView"
+import postData from './mock-data'
+
+const SearchPage = () => {
+    const [searchTerm, setSeachTerm] = useState('')
+
```
You are missing an import here for useState

nzbasic commented on frontend/src/pages/post/SearchPage.jsx:
```diff
@@ -0,0 +1,25 @@
+            return val
+            }
+        }).map((val, key)=> {
+            return (<div className="content" key={key}>{" "} <p>{val.author}</p><p>{val.content}</p><br></br></div>);
+          })}
```
You should return a `<Post key={key} data={val} />` here instead, it is a component we use for displaying post data

jmor528 committed:
[#100] - GET /posts/:id/interactions (#122)

* Add interaction functions for getting the users that liked and shared the post

* Testing the get posts interaction endpoint

* Classes have default export and all asynchronous loops now use promise.all

* merge with upstream main

* Create helper functions to create users,posts,etc for unit tests

* Problem with serverInstance listening to 8000, using server 8080 for all tests

nzbasic reviewed (changes requested):
Please have a look at our react conventions guide https://github.com/SE701Team2/Updog/wiki/React-Conventions 

Components/Pages should be in their own folders, so you should move this to a search folder inside posts.

JL-2644 committed:
[#112] Add all missing user and post routes to swagger (#126)

* Add all missing user and post routes to swagger
- users: modifyUser, deleteUser, activity, follow, feed, and notifications
- posts: like, share, and interactions

hmit106 committed:
[#133] Fix typo in LandingView.jsx (#134)

nzbasic committed:
[#106] - Replace mock data with backend connection (#124)

* refactor to use generic request method

* add username to localstorage on login

* update routes

* hookup backend to header

* hookup backend to dashboard

* fix styling issues

* user page backend connection

* fix api request url

* post page backend connection

* post backend connection

* update to use post id

* rename to jsx, fix eslint

* follows page to backend

* implement follow/unfollow

* fix model not correct

* implement follow/unfollow with logic

* add link to post on replies

* connect settings to backend

* clean up destructuring

* fix jwt not expiring

* use correct route for feed

* connect notifications to backend

* update logo

* add link to new post

* use correct settings route

* fix react error

* fix race condition

* use correct unix time

* force commit current md

* remove mock data

hmit106 committed:
[#123] Add e2e testing framework (#142)

* Add placeholder e2e test

* Add and install Cypress

* Add Cypress plugin to ESLint

* Add script to run cypress with `npm run e2e`

* Add baseUrl to cypress.json

MarkGuo2000 committed:
Bug fixs

Changed typo in Router.jsx, moved folder for SearchPage.jsx. Used Post key={key} data={val} as output

MarkGuo2000 committed:
Resolve conflict on Json files

htom380 commented:
James has refactored the code to use data from the DB rather than mock-data now, you'll need to incorporate that change into your branch before this can be merged I think

harristomy committed:
[#144] Added ESLint Cypress plugin to global (#145)

harristomy commented on frontend/src/pages/post/Search/SearchPage.jsx:
```diff
@@ -0,0 +1,26 @@
+import { useParams } from "react-router-dom"
+import postData from '../mock-data'
```
James has refactored the code to use data from the DB rather than mock-data now, you'll need to incorporate that change into your branch before this can be merged I think

harristomy committed:
[#141] Added CI Lint checking for ESLint and Prettier (#146)

* Added CI Lint checking

* Updated CONTRIBUTING.md to show CI Lint step

* Typo in MD fix

MarkGuo2000 committed:
Use Data from backend for search

nzbasic commented on frontend/src/pages/post/Search/SearchPage.jsx:
```diff
@@ -0,0 +1,26 @@
+    return (
+        <div className="Search">
+            <TextField id="standard-basic" label="Standard" variant="standard" size="normal" placeholder="Seach..." onChange={event => {setSeachTerm(event.target.value);}} />
+            <br></br>
+            {Post.filter((val) => {
+            if (searchTerm=="") {
+            return val
```
You can't use filter on an imported component, I am not sure what you are trying to do here. The page will not run

MarkGuo2000 replied:
My bad I assumed Post is structured the same as the mockdata so I used the same filter. Mind giving me a tip on how to get to the post contents(text) thank you.

nzbasic commented on frontend/src/Router.jsx:
```diff
@@ -33,6 +35,7 @@ const Router = () => {
         <Route path="/notifications" element={<Notifications />} />
         <Route path="/settings" element={<ProfileSettings />} />
         <Route path="/new-post" element={<PostComposer />} />
+        <Route path="/search-page" element={<SearchPage/>} />
```
This should just be /search

nzbasic commented on frontend/src/pages/post/Search/SearchPage.jsx:
```diff
@@ -0,0 +1,26 @@
+            <TextField id="standard-basic" label="Standard" variant="standard" size="normal" placeholder="Seach..." onChange={event => {setSeachTerm(event.target.value);}} />
+            <br></br>
+            {Post.filter((val) => {
+            if (searchTerm=="") {
+            return val
+            } else if ((val.content.toLowerCase().includes(searchTerm.toLowerCase())) || (val.author.toLowerCase().includes(searchTerm.toLowerCase()))) {
+            return val
```
These returns should be returning true to work with filter

MarkGuo2000 replied:
was using "val.content.toLowerCase().includes(searchTerm.toLowerCase()" to narrowdown all the posts with containing the keywords. If the searchTerm is nothing it suppose to just return all present posts. Probably this didn't work because the mockdata and back end database was arranged differently I think.

nzbasic reviewed (changes requested):
The files should be under their own directory e.g. /pages/search, not /pages/post/Search

The page does not render run and produces a lot of errors in console, probably due to the Post.filter. If you want, you can create some empty state like 
```js
const [posts, setPosts] = useState([]) 
```
and use that to filter from 


The linting check has failed, you will need to fix the prettier/eslint errors. You should read our CONTRIBUTING.md file in the repo to see how to configure your vscode to show you the errors and apply styling.

You should also add a search.module.scss file to the search directory so you can apply styles to your component.

MarkGuo2000 committed:
Add header and footer, and change file location

MarkGuo2000 committed:
Change Filtering Method

scorpionknifes commented:
PR is stale with too many merge conflicts. 

A2 has decided to abandon this PR
